agent/provider/src/rules.rs

```diff
@@ -62,30 +65,84 @@ impl RulesManager {
 
     pub fn remove_dangling_rules(&self) -> Result<()> {
         let mut deleted_partner_rules = vec![];
+        let mut deleted_audited_payload_rules = vec![];
 
         let keystore_certs = self.keystore.list_ids();
 
+        let mut rulestore = self.rulestore.config.write().unwrap();
+
+        rulestore.outbound.partner.retain(|cert_id, _| {
+            keystore_certs
+                .contains(cert_id)
+                .not()
+                .then(|| deleted_partner_rules.push(cert_id.clone()))
+                .is_none()
+        });
+
+        rulestore.outbound.audited_payload.retain(|cert_id, _| {
+            keystore_certs
+                .contains(cert_id)
+                .not()
+                .then(|| deleted_audited_payload_rules.push(cert_id.clone()))
+                .is_none()
+        });
+
+        drop(rulestore);
```

Comment:
[NITPICK]: I would prefer putting this stuff in separate scope to drop guard (as it is done in such way in yagna normally)

Reply:
Refactored.